core/src/test/java/hudson/UtilTest.java

```diff
@@ -490,9 +494,13 @@ private void lockFileForDeletion(File f) throws IOException, InterruptedExceptio
         // On unix, can't use "chattr +u" because ext fs ignores it.
         // On Windows, we can't delete files that are open for reading, so we use that.
         assert Functions.isWindows();
-        final InputStream s = new FileInputStream(f);
+        final FileChannel channel = FileChannel.open(f.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
```

Comment:
Though it is correct change for file locking API, it follows the presumption that All windows FileSystems support locking. And it is not correct. If the locking is not supported, IIRC the lock() call throws the IOException, which goes outside the method. Maybe it would be better to check if locking is supported in assert/assume